src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -101,7 +109,16 @@ public void save(AddressBook addressBook) throws StorageOperationException {
         } catch (JAXBException jaxbe) {
             throw new StorageOperationException("Error converting address book into storage format");
         }
-    }
+	}
+
+	/*
+	 * Check to see if storage file was deleted
+	 */
+    private void checkStorageFile() throws StorageOperationException {
+		if(!path.toFile().exists()) {
```

Comment:
- Good try. But may be throw the exception after you recovered the file.

- Where did you handle this exception? Best place would be the executeCommand() Main.java